src/NLog/Config/LoggingConfigurationParser.cs

```diff
@@ -168,10 +256,13 @@ protected virtual bool ParseNLogSection(ILoggingConfigurationElement configSecti
                     ParseTargetsElement(configSection);
                     return true;
             }
+
             return false;
         }
 
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom", Justification = "Need to load external assembly.")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability",
+            "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom",
+            Justification = "Need to load external assembly.")]
         private void ParseExtensionsElement(ILoggingConfigurationElement extensionsElement, string baseDirectory)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 27 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)